_posts/2025-02-07-this-week-in-rails.markdown

```diff
@@ -0,0 +1,49 @@
+---
+layout: post
+title: "Support joins in update_all for PG and SQLite and more"
+categories: news
+author: Wojtek
+og_image: assets/images/this-week-in-rails.png
+published: true
+date: 2025-02-07
+---
+
+
+Hi, [Wojtek](https://x.com/morgoth85) here presenting you this week's changes in the Rails codebase.
+
+[Clio joins the Rails Foundation](https://rubyonrails.org/2025/2/3/clio-joins-rails-foundation)  
+We’re excited to welcome [Clio](https://www.clio.com/) as the newest Contributing Member of the Rails Foundation.
+
+[Support joins in update_all for Postgresql and SQlite](https://github.com/rails/rails/pull/53950)  
+Previously when generating update queries with one or more JOIN clauses,
+Active Record would use a sub query which would prevent to reference the joined
+tables in the *SET* clause, for instance:
+
+```ruby
+Comment.joins(:post).update_all("title = posts.title")
+```
+
+This is now supported as long as the relation doesn't also use a *LIMIT*, *ORDER* or *GROUP BY* clause. This was supported by the MySQL adapter for a long time.
+
+[Rate limit password resets in auth generator](https://github.com/rails/rails/pull/54442)  
+This helps mitigate abuse from attackers spamming the password reset form.
+
+[Enable statement-cached queries to be retryable](https://github.com/rails/rails/pull/54436)  
+Enable automatically retrying idempotent association queries on connection errors.
+
+[Add allow_retry to sql.active_record](https://github.com/rails/rails/pull/54454)  
+This enables identifying queries which queries are automatically retryable on connection errors.
+
+[Migrate multiple DBs with pending migration action](https://github.com/rails/rails/pull/54456)  
+Fix migrating multiple databases with *ActiveRecord::PendingMigration* action.
+  
+[Speed up Docker build](https://github.com/rails/rails/pull/54428)  
+By removing recursive *chown* command, default docker build should now be a bit faster.
```

Comment:
We can expand here to mention its speeding up the default docker template? Else its not clear which docker build we are talking about